src/Workspaces/Core/Portable/FindSymbols/FindReferences/BaseTypeFinder.cs

```diff
@@ -2,71 +2,61 @@
 
 using System.Collections.Immutable;
 using System.Threading;
-using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.FindSymbols.FindReferences
 {
     internal static partial class BaseTypeFinder
     {
-        public static async Task<ImmutableArray<SymbolAndProjectId>> FindBaseTypesAndInterfacesAsync(
-            INamedTypeSymbol type, Project project, CancellationToken cancellationToken)
-        {
-            var typesAndInterfaces = FindBaseTypesAndInterfaces(type);
-            return await ConvertToSymbolAndProjectIdsAsync(typesAndInterfaces.CastArray<ISymbol>(), project, cancellationToken).ConfigureAwait(false);
-        }
+        public static ImmutableArray<ISymbol> FindBaseTypesAndInterfaces(INamedTypeSymbol type)
+            => FindBaseTypes(type).AddRange(type.AllInterfaces).CastArray<ISymbol>();
 
-        public static async Task<ImmutableArray<SymbolAndProjectId>> FindOverriddenAndImplementedMembersAsync(
+        public static ImmutableArray<ISymbol> FindOverriddenAndImplementedMembers(
             ISymbol symbol, Project project, CancellationToken cancellationToken)
         {
             var solution = project.Solution;
-            var results = ArrayBuilder<SymbolAndProjectId>.GetInstance();
-            var interfaceImplementations = ArrayBuilder<ISymbol>.GetInstance();
+            var results = ArrayBuilder<ISymbol>.GetInstance();
 
             // This is called for all: class, struct or interface member.
-            interfaceImplementations.AddRange(symbol.ExplicitOrImplicitInterfaceImplementations());
+            results.AddRange(symbol.ExplicitOrImplicitInterfaceImplementations());
 
             // The type scenario. Iterate over all base classes to find overridden and hidden (new/Shadows) methods.
             foreach (var type in FindBaseTypes(symbol.ContainingType))
             {
                 foreach (var member in type.GetMembers(symbol.Name))
                 {
-                    var sourceMember = await SymbolFinder.FindSourceDefinitionAsync(
```

Comment:
Now that this returns metadata symbols, do we expose this anywhere as a public API that this is a breaking change of?

Reply:
We didn't change FindSourceDefinitionAsync. Only in case, where we cannot get sources by it; we go to metadata.

---
In reply to: [335223864](https://github.com/dotnet/roslyn/pull/39055#discussion_r335223864) [](ancestors = 335223864)

Reply:
Methods changed belong to internal classes and should not be involved in public APIs

---
In reply to: [335726193](https://github.com/dotnet/roslyn/pull/39055#discussion_r335726193) [](ancestors = 335726193,335223864)